config/homebin/phpswap

```diff
@@ -0,0 +1,173 @@
+			--pid ${PHPFPM_PIDFILE}
+		if [[ $? != "0" ]] ; then
+			echo "php-fpm start failed."
+		fi
+	fi
+}
+
+function brew_fpm_off {
+	# Unfortunately it looks the the built in PHPBrew runner doesn't work with sudo so
+	# we'll have to mimick the functionality
+	if [[ $PHPFPM_INUSE =~ ^[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+		PHPFPM_PIDFILE=${PHPBREW_ROOT}/php/php-${PHPFPM_INUSE}/var/run/php-fpm.pid
+		if [[ -e ${PHPFPM_PIDFILE} ]] ; then
+			echo "Stopping phpbrew's php-fpm..."
+			sudo kill $(cat ${PHPFPM_PIDFILE})
+			sudo rm -f ${PHPFPM_PIDFILE}
+		fi
+	fi
+}
+
+function brew_fpm_restart {
+	brew_fpm_off
+	brew_fpm_on $PHPFPM_INUSE
+}
+
+# Allow users to turn PHPBrew off
+if [[ $1 == "system" ]]; then
+	brew_fpm_off
+	phpbrew off
+	sudo service php5-fpm stop > /dev/null 2>&1
+	sudo service php5-fpm start
+	echo "$(tput setaf 6)PHP switched back to system version. PHPBrew is off.$(tput sgr 0)"
+	exit 0
+fi
+
+# Allow users to easily restart PHPBrew's FPM
+if [[ $1 == "fpm-restart" ]]; then
+	brew_fpm_restart
+	echo "$(tput setaf 6)PHPBrew's FPM restarted.$(tput sgr 0)"
+	exit 0
+fi
+
+# Check if -s was passed so we know if we're going to 'use' or 'switch'
+SWAP_TYPE=use
+while getopts 's' flag; do
+  case "${flag}" in
+    s) SWAP_TYPE=switch; shift;;
+    *) echo "Usage:  phpswap [-s] PHP_VERSION"; exit 1;;
+  esac
+done
+
+# Check for version number
+if ! [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+	echo "$(tput setaf 1)Please enter a proper php version number$(tput sgr 0)"
+	exit 1
+fi
+
+PHP_VER=$1
+BREW_USE=`phpbrew $SWAP_TYPE $PHP_VER | tr -cd [:print:]`
+BREW_VERSION_AVAIL=`phpbrew list | grep $PHP_VER | wc -l`
+
+# Check for faulty phpbrew config
+if [[ $BREW_USE == *"should not see this"* ]]; then
+	echo "$(tput setaf 1)ERROR: PHPBrew misconfigured$(tput sgr 0)"
+	echo "$BREW_USE"
+	exit 1
+fi
+
+# Make sure the version exists on the local system
+if [[ $BREW_USE == *"not exists"* || $BREW_VERSION_AVAIL == 0 ]]; then
+	# We need to install the new phpversion
+	phpbrew update
+	echo "$(tput setaf 6)PHP version $PHP_VER not installed. Building from source/installing... [this may take a few mintues]$(tput sgr 0)"
+	VERSION_INSTALL=$(phpbrew install $PHP_VER +default +openssl +cgi +mb +mcrypt +mysql +pdo +gd +json +readline +fpm | tee /dev/tty)
+
+	# Make sure the version was found within PHPBrew
+	VER_NOT_FOUND=`echo $VERSION_INSTALL | grep -i "not found"`
+	if [[ ${#VER_NOT_FOUND} > 0 ]]; then
+		echo "$(tput setaf 1)PHP Version $PHP_VER not found. Exiting.$(tput sgr 0)"
+		exit 1
+	fi
+	# Yay, it was a valid version. Lets see if it installed/compiled correctly
+	if [ "$?" != "0" ]; then
+		echo "$(tput setaf 1)Install/compilation failed for php version $PHP_VER$(tput sgr 0)"
+		exit 1
+	fi
+
+	# Make sure Ubuntu's php-fpm is stopped
+	sudo service php5-fpm stop
+
+	# Switch the PHP version
+	phpbrew $SWAP_TYPE $PHP_VER
+	# Install extensions
+	phpbrew ext install memcache
+	phpbrew ext install imagick
```

Comment:
For PHP 7, right now, imagick stable fails to install.  imagick 3.4.0RC6 seems to install.